sympy/core/symbol.py

```diff
@@ -132,11 +143,11 @@ def _uniquely_named_symbol(xname, exprs=(), compare=str, modify=None, **assumpti
         exprs = [exprs]
     syms = set().union(*[e.free_symbols for e in exprs])
```

Comment:
@smichr

`free_symbols` does not include dummy symbols in `Subs`, `Integral`, etc.  
IMO this should be changed to
```python
syms = set()
for e in exprs:
    syms = syms.union([a for a in e.atoms() if a.is_Symbol])
```
How do you think?

Reply:
Agree. I wonder if function names should be included.
```
names = set().union(*([i.name for e in exprs for i in e.atoms(Symbol)] +
    [getattr(i.func, 'name', str(i.func)) for e in exprs for i in e.atoms(Function)])))
```

Reply:
Got it done. Thanks.